x/programs/rust/wasmlanche-sdk/src/state.rs

```diff
@@ -85,3 +113,55 @@ impl Key {
         Self(bytes)
     }
 }
+
+mod host {
+    use super::{BorshSerialize, Key, Program};
+    use crate::{memory::to_host_ptr, state::Error};
+
+    #[link(wasm_import_module = "state")]
+    extern "C" {
+        #[link_name = "put"]
+        fn _put(caller: i64, key: i64, value: i64) -> i64;
+
+        #[link_name = "get"]
+        fn _get(caller: i64, key: i64) -> i64;
+
+        #[link_name = "delete"]
+        fn _delete(caller: i64, key: i64) -> i64;
+    }
+
+    /// Persists the bytes at `value_ptr` to the bytes at key ptr on the host storage.
+    pub(super) unsafe fn put_bytes<V>(caller: &Program, key: &Key, value: &V) -> Result<(), Error>
```

Comment:
just a question to gain more context: why do all these functions have `unsafe`?

Reply:
That's a great question. I think there was a point in time when these functions were doing something unsafe beyond calling FFI functions. `unsafe` is supposed to be used to mark functions that could lead to undefined behaviour (UB), but I think since we've introduced borsh-serialization, there's less of that here. 

Marking a function `unsafe` forces you to use an unsafe block, `unsafe { /* function call goes here */ }`, to call it. Usually, that means you have to define your safety parameters/invariants outside of the function and add necessary tests to guarantee safety. 

Make sense? This was a great question because you've made me realize that I probably need to clean this code up. But I'm going to do that in another PR